commands.py

```diff
@@ -277,3 +277,212 @@ def run_discord_bot():
+                    RwrAccount.username.in_(all_player_names)
+                ).all()
+
+                rwr_accounts_by_username = {rwr_account.username: rwr_account for rwr_account in existing_rwr_accounts}
+
+                # Remove players already having an RWR account and that are already up-to-date in the DB to prevent saving duplicate stats
+                for player in players:
+                    username = player[0] # FIXME Username isn't properly encoded, try to decode it to unicode
+
+                    if username in rwr_accounts_by_username and rwr_accounts_by_username[username].updated_at.floor('day') >= now:
+                        players.remove(player)
+
+                # Create RWR accounts if they do not exists / touch the updated_at timestamp if they exists
+                for player in players:
+                    username = player[0] # FIXME Username isn't properly encoded, try to decode it to unicode
+
+                    if username not in rwr_accounts_by_username:
+                        rwr_account = RwrAccount()
+
+                        rwr_account.username = username
+                        rwr_account.type = RwrAccountType.INVASION
+
+                        rwr_accounts_by_username[username] = rwr_account
+                    else:
+                        rwr_account = rwr_accounts_by_username[username]
+
+                        rwr_account.updated_at = arrow.utcnow().floor('minute')
+
+                    db.session.add(rwr_account)
+
+                db.session.commit()
+
+                # Save current stats of the RWR accounts (if their respective stats changed)
+                for player in players:
+                    username = player[0] # FIXME Username isn't properly encoded, try to decode it to unicode
+
+                    rwr_account_stat = RwrAccountStat()
+
+                    rwr_account_stat.leaderboard_position = leaderboard_position
+                    rwr_account_stat.xp = int(player[1])
+                    rwr_account_stat.kills = int(player[3])
+                    rwr_account_stat.deaths = int(player[4])
+                    rwr_account_stat.time_played = int(player[2]) * 60
+                    rwr_account_stat.longest_kill_streak = int(player[5])
+                    rwr_account_stat.targets_destroyed = int(player[6])
+                    rwr_account_stat.vehicles_destroyed = int(player[7])
+                    rwr_account_stat.soldiers_healed = int(player[8])
+                    rwr_account_stat.teamkills = int(player[9])
+                    rwr_account_stat.distance_moved = round(int(player[10]) / 1000, 1)
+                    rwr_account_stat.shots_fired = int(player[10])
+                    rwr_account_stat.throwables_thrown = int(player[12])
+                    rwr_account_stat.created_at = created_at
+                    rwr_account_stat.rwr_account_id = rwr_accounts_by_username[username].id
+
+                    if rwr_account_stat.can_be_saved:
```

Comment:
Change logic here to save a big number of DB queries: create all the players `RwrAccountStat` objects, then query one time the DB to check with all the hashes.